hirzenbach-bot/vulgar_fraction.py

```diff
@@ -0,0 +1,132 @@
+from dataclasses import dataclass
+from typing import Optional, Callable, Iterable
+
+_FRACTION_SLASH = '\u2044'
+
+
+@dataclass(frozen=True, repr=False)
+class VulgarFraction:
+    """
+    >>> str(VulgarFraction(-1234567890, -1234567890))
+    '⁻¹²³⁴⁵⁶⁷⁸⁹⁰⁄₋₁₂₃₄₅₆₇₈₉₀'
+    >>> str(VulgarFraction(3, 4))
+    '¾'
+    """
+    nominator: int
+    denominator: int
+
+    def __repr__(self):
+        return _single_character_fraction(self) or _format_vulgar_fraction(self.nominator, self.denominator)
+
+
+def _format_vulgar_fraction(nominator: int, denominator: int) -> str:
+    return f'{_format_number(nominator, _superscript)}{_FRACTION_SLASH}{_format_number(denominator, _subscript)}'
+
+
+def _format_number(n: int, format_digit: Callable[[str], str]) -> str:
+    return ''.join((format_digit(digit) for digit in _digits(n)))
+
+
+def _digits(n: int) -> Iterable[str]:
+    return str(n)
+
+
+def _superscript(character: str) -> str:
+    match character:
+        case '-':
```

Comment:
no `+`?

Reply:
We never format integers with explicit leading +, so not necessary